master/pyext/src/validation/excludedvolume.py

```diff
@@ -1,125 +1,165 @@
+            df = self.get_xyzr(model)
+            excluded_volume['Excluded Volume Satisfaction'].append(
+                round(self.get_violation_percentage(df, self.get_violation_dict(df)), 2))
+            excluded_volume['Number of violations'].append(
+                sum(list(self.get_violation_dict(df).values())))
+        # open(os.path.join(os.getcwd(), self.resultpath, self.ID+'exv.txt'), 'w+')
         return excluded_volume
 
-    def get_exc_vol_for_models_normalized(self,model_dict:dict)->dict:
-        excluded_volume={'Models':[],'Excluded Volume Satisfaction':[]}
+    def get_exc_vol_for_models_normalized(self, model_dict: dict) -> dict:
+        excluded_volume = {'Models': [], 'Excluded Volume Satisfaction': []}
         for indx, model in model_dict.items():
             excluded_volume['Models'].append(indx)
-            df=self.get_xyzr(model)
-            satisfaction=self.get_violation_normalized(df,self.get_violation_dict(df))
-            excluded_volume['Excluded Volume Satisfaction'].append(round(satisfaction,2))
+            df = self.get_xyzr(model)
+            satisfaction = self.get_violation_percentage(
+                df, self.get_violation_dict(df))
+            excluded_volume['Excluded Volume Satisfaction'].append(
+                round(satisfaction, 2))
         return excluded_volume
 
-    def get_exc_vol_given_sphere_parallel(self,sphere_list:list)->(float,int):
+    def get_exc_vol_given_sphere_parallel(self, sphere_list: list) -> (float, int):
         """ """
-        df=self.get_xyzr(sphere_list)
-        violation_dict=self.get_violation_dict(df)
-        satisfaction=round(self.get_violation_percentage(df,violation_dict),2)
-        violations=sum(list(violation_dict.values()))
-        return (satisfaction,violations)
+        df = self.get_xyzr(sphere_list)
+        violation_dict = self.get_violation_dict(df)
+        satisfaction = round(
+            self.get_violation_percentage(df, violation_dict), 2)
+        violations = sum(list(violation_dict.values()))
+        return (satisfaction, violations)
 
-    def run_exc_vol_parallel(self,model_dict:dict)->dict:
+    def run_exc_vol_parallel(self, model_dict: dict) -> dict:
         """ get exc vol info in parallel """
-        list_of_sphere_list=list(model_dict.values())
-        pool=mp.Pool(processes=len(list(model_dict.keys())))
-        complete_list=pool.map(self.get_exc_vol_given_sphere_parallel,list(model_dict.values()))
-        excluded_volume={'Models':list(model_dict.keys()),'Excluded Volume Satisfaction':[i[0] for i in complete_list], 'Number of violations':[i[1] for i in complete_list]}
-        f_exv=open(os.path.join(os.getcwd(),'../static/results/',self.ID+'exv.txt'),'w+')
+        # list_of_sphere_list=list(model_dict.values())
+        filename = os.path.join(os.getcwd(),
+                                self.resultpath, self.ID+'exv.txt')
```

Comment:
Are you trying to make a relative path into an absolute one? If so, use `os.path.abspath`.

Reply:
I prefer to leave this as is, because I can change resultpath (directory where I want to store results).